src/AppRoot.tsx

```diff
@@ -497,11 +497,40 @@ class AppRoot extends React.Component<Props, State> {
         }
       )
     }
+
+    /* istanbul ignore else */
+    if (!this.onDeviceChangeListener) {
+      const { hardware } = this.props
+      ;(async () => {
+        // get initial status
+        const accessibleControllerStatus = await hardware.readAccesssibleControllerStatus()
+
+        this.setState({
+          hasAccessibleControllerAttached: accessibleControllerStatus.connected,
+        })
+
+        // watch for changes
+        this.onDeviceChangeListener = hardware.onDeviceChange.add(
```

Comment:
This won’t be set by the time the method exits, which means this could be added twice. I should address that.